File: android/src/main/kotlin/com/example/konnek_flutter/response/GetConversationResponseModel.kt
package com.example.konnek_flutter.response

import kotlinx.parcelize.Parcelize
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.RawValue

@Parcelize
data class GetConversationResponseModel(

	@field:SerializedName("data")
	val data: DataGetConversation? = null,

	@field:SerializedName("meta")
	val meta: MetaGetConversation? = null
) : Parcelable

@Parcelize
data class AgentGetConversation(

	@field:SerializedName("online_status")
	val onlineStatus: Int? = null,

	@field:SerializedName("customer_channel")
	val customerChannel: String? = null,

	@field:SerializedName("description")
	val description: String? = null,

	@field:SerializedName("division_id")
	val divisionId: String? = null,

	@field:SerializedName("avatar")
	val avatar: String? = null,

	@field:SerializedName("tags")
	val tags: String? = null,

	@field:SerializedName("is_blocked")
	val isBlocked: Boolean? = null,

	@field:SerializedName("phone")
	val phone: String? = null,

	@field:SerializedName("name")
	val name: String? = null,

	@field:SerializedName("id")
	val id: String? = null,

	@field:SerializedName("email")
	val email: String? = null,

	@field:SerializedName("username")
	val username: String? = null,

	@field:SerializedName("status")
	val status: Int? = null
) : Parcelable

@Parcelize
data class CloseUserGetConversation(

	@field:SerializedName("online_status")
	val onlineStatus: Int? = null,

	@field:SerializedName("customer_channel")
	val customerChannel: String? = null,

	@field:SerializedName("description")
	val description: String? = null,

	@field:SerializedName("division_id")
	val divisionId: String? = null,

	@field:SerializedName("avatar")
	val avatar: String? = null,

	@field:SerializedName("tags")
	val tags: String? = null,

	@field:SerializedName("is_blocked")
	val isBlocked: Boolean? = null,

	@field:SerializedName("phone")
	val phone: String? = null,

	@field:SerializedName("name")
	val name: String? = null,

	@field:SerializedName("id")
	val id: String? = null,

	@field:SerializedName("email")
	val email: String? = null,

	@field:SerializedName("username")
	val username: String? = null,

	@field:SerializedName("status")
	val status: Int? = null
) : Parcelable

@Parcelize
data class SessionGetConversation(

	@field:SerializedName("room_id")
	val roomId: String? = null,

	@field:SerializedName("agent_user_id")
	val agentUserId: String? = null,

	@field:SerializedName("last_agent_chat_time")
	val lastAgentChatTime: String? = null,

	@field:SerializedName("agent")
	val agent: AgentGetConversation? = null,

	@field:SerializedName("bot_status")
	val botStatus: Boolean? = null,

	@field:SerializedName("division_id")
	val divisionId: String? = null,

	@field:SerializedName("open_time")
	val openTime: String? = null,

	@field:SerializedName("close_time")
	val closeTime: String? = null,

	@field:SerializedName("close_user")
	val closeUser: CloseUserGetConversation? = null,

	@field:SerializedName("queue_time")
	val queueTime: String? = null,

	@field:SerializedName("first_response_time")
	val firstResponseTime: String? = null,

	@field:SerializedName("session_priority_level_id")
	val sessionPriorityLevelId: String? = null,

	@field:SerializedName("open_by")
	val openBy: String? = null,

	@field:SerializedName("id")
	val id: String? = null,

	@field:SerializedName("categories")
	val categories: String? = null,

	@field:SerializedName("assign_time")
	val assignTime: String? = null,

	@field:SerializedName("close_by")
	val closeBy: String? = null,

	@field:SerializedName("handover_by")
	val handoverBy: String? = null,

	@field:SerializedName("status")
	val status: Int? = null
) : Parcelable

@Parcelize
data class UserGetConversation(

	@field:SerializedName("online_status")
	val onlineStatus: Int? = null,

	@field:SerializedName("customer_channel")
	val customerChannel: String? = null,

	@field:SerializedName("description")
	val description: String? = null,

	@field:SerializedName("division_id")
	val divisionId: String? = null,

	@field:SerializedName("avatar")
	val avatar: String? = null,

	@field:SerializedName("tags")
	val tags: String? = null,

	@field:SerializedName("is_blocked")
	val isBlocked: Boolean? = null,

	@field:SerializedName("phone")
	val phone: String? = null,

	@field:SerializedName("name")
	val name: String? = null,

	@field:SerializedName("id")
	val id: String? = null,

	@field:SerializedName("email")
	val email: String? = null,

	@field:SerializedName("username")
	val username: String? = null,

	@field:SerializedName("status")
	val status: Int? = null
) : Parcelable

@Parcelize
data class DataGetConversation(

	@field:SerializedName("expire")
	val expire: String? = null,

	@field:SerializedName("conversations")
	val conversations: List<ConversationsItemGetConversation?>? = null,

	@field:SerializedName("room")
	val room: RoomGetConversation? = null,

	@field:SerializedName("token")
	val token: String? = null
) : Parcelable

@Parcelize
data class RoomGetConversation(

	@field:SerializedName("customer_description")
	val customerDescription: String? = null,

	@field:SerializedName("last_customer_message_time")
	val lastCustomerMessageTime: String? = null,

	@field:SerializedName("open_time")
	val openTime: String? = null,

	@field:SerializedName("channel_code")
	val channelCode: String? = null,

	@field:SerializedName("customer_username")
	val customerUsername: String? = null,

	@field:SerializedName("open_by")
	val openBy: String? = null,

	@field:SerializedName("assign_time")
	val assignTime: String? = null,

	@field:SerializedName("categories")
	val categories: String? = null,

	@field:SerializedName("id")
	val id: String? = null,

	@field:SerializedName("close_by")
	val closeBy: String? = null,

	@field:SerializedName("handover_by")
	val handoverBy: String? = null,

	@field:SerializedName("agent_user_id")
	val agentUserId: String? = null,

	@field:SerializedName("last_agent_chat_time")
	val lastAgentChatTime: String? = null,

	@field:SerializedName("company_id")
	val companyId: String? = null,

	@field:SerializedName("bot_status")
	val botStatus: Boolean? = null,

	@field:SerializedName("division_id")
	val divisionId: String? = null,

	@field:SerializedName("session_id")
	val sessionId: String? = null,

	@field:SerializedName("send_outbound_flag")
	val sendOutboundFlag: Boolean? = null,

	@field:SerializedName("close_time")
	val closeTime: String? = null,

	@field:SerializedName("queue_time")
	val queueTime: String? = null,

	@field:SerializedName("customer_avatar")
	val customerAvatar: String? = null,

	@field:SerializedName("first_response_time")
	val firstResponseTime: String? = null,

	@field:SerializedName("customer_email")
	val customerEmail: String? = null,

	@field:SerializedName("customer_name")
	val customerName: String? = null,

	@field:SerializedName("customer_id")
	val customerId: String? = null,

	@field:SerializedName("window_messaging")
	val windowMessaging: String? = null,

	@field:SerializedName("status")
	val status: Int? = null
) : Parcelable

@Parcelize
data class MetaGetConversation(

	@field:SerializedName("next_page")
	val nextPage: Boolean? = null,

	@field:SerializedName("log_id")
	val logId: String? = null,

	@field:SerializedName("per_page")
	val perPage: Int? = null,

	@field:SerializedName("code")
	val code: Int? = null,

	@field:SerializedName("total_count")
	val totalCount: Int? = null,

	@field:SerializedName("message")
	val message: String? = null,

	@field:SerializedName("prev_page")
	val prevPage: Boolean? = null,

	@field:SerializedName("errors")
	val errors: @RawValue Any? = null,

	@field:SerializedName("current_page")
	val currentPage: Int? = null,

	@field:SerializedName("page_count")
	val pageCount: Int? = null,

	@field:SerializedName("status")
	val status: Boolean? = null
) : Parcelable

@Parcelize
data class ConversationsItemGetConversation(

	@field:SerializedName("room_id")
	val roomId: String? = null,

	@field:SerializedName("from_type")
	val fromType: String? = null,

	@field:SerializedName("session")
	val session: SessionGetConversation? = null,

	@field:SerializedName("session_id")
	val sessionId: String? = null,

	@field:SerializedName("created_at")
	val createdAt: String? = null,

	@field:SerializedName("message_id")
	val messageId: String? = null,

	@field:SerializedName("type")
	val type: String? = null,

	@field:SerializedName("created_by")
	val createdBy: String? = null,

	@field:SerializedName("message_time")
	val messageTime: String? = null,

	@field:SerializedName("reply_id")
	val replyId: String? = null,

	@field:SerializedName("user_id")
	val userId: String? = null,

	@field:SerializedName("payload")
	val payload: String? = null,

	@field:SerializedName("unix_msg_time")
	val unixMsgTime: Long? = null,

	@field:SerializedName("seq_id")
	val seqId: Int? = null,

	@field:SerializedName("id")
	val id: String? = null,

	@field:SerializedName("text")
	val text: String? = null,

	@field:SerializedName("user")
	val user: UserGetConversation? = null,

	@field:SerializedName("status")
	val status: Int? = null
) : Parcelable

File: ios/Classes/response/GetConversationResponseModel.swift
//import Foundation
//
//// MARK: - Root Response
//struct GetConversationResponseModel: Codable {
//    let data: DataGetConversation?
//    let meta: MetaGetConversation?
//}
//
//// MARK: - Data Section
//struct DataGetConversation: Codable {
//    let expire: String?
//    let conversations: [ConversationsItemGetConversation?]?
//    let room: RoomGetConversation?
//    let token: String?
//}
//
//// MARK: - Meta Section
//struct MetaGetConversation: Codable {
//    let nextPage: Bool?
//    let logId: String?
//    let perPage: Int?
//    let code: Int?
//    let totalCount: Int?
//    let message: String?
//    let prevPage: Bool?
//    let errors: AnyCodable?
//    let currentPage: Int?
//    let pageCount: Int?
//    let status: Bool?
//
//    enum CodingKeys: String, CodingKey {
//        case nextPage = "next_page"
//        case logId = "log_id"
//        case perPage = "per_page"
//        case code
//        case totalCount = "total_count"
//        case message
//        case prevPage = "prev_page"
//        case errors
//        case currentPage = "current_page"
//        case pageCount = "page_count"
//        case status
//    }
//}
//
//// MARK: - Conversations Item
//struct ConversationsItemGetConversation: Codable {
//    let roomId: String?
//    let fromType: String?
//    let session: SessionGetConversation?
//    let sessionId: String?
//    let createdAt: String?
//    let messageId: String?
//    let type: String?
//    let createdBy: String?
//    let messageTime: String?
//    let replyId: String?
//    let userId: String?
//    let payload: String?
//    let unixMsgTime: Int64?
//    let seqId: Int?
//    let id: String?
//    let text: String?
//    let user: UserGetConversation?
//    let status: Int?
//
//    enum CodingKeys: String, CodingKey {
//        case roomId = "room_id"
//        case fromType = "from_type"
//        case session
//        case sessionId = "session_id"
//        case createdAt = "created_at"
//        case messageId = "message_id"
//        case type
//        case createdBy = "created_by"
//        case messageTime = "message_time"
//        case replyId = "reply_id"
//        case userId = "user_id"
//        case payload
//        case unixMsgTime = "unix_msg_time"
//        case seqId = "seq_id"
//        case id
//        case text
//        case user
//        case status
//    }
//}
//
//// MARK: - Room
//struct RoomGetConversation: Codable {
//    let customerDescription: String?
//    let lastCustomerMessageTime: String?
//    let openTime: String?
//    let channelCode: String?
//    let customerUsername: String?
//    let openBy: String?
//    let assignTime: String?
//    let categories: String?
//    let id: String?
//    let closeBy: String?
//    let handoverBy: String?
//    let agentUserId: String?
//    let lastAgentChatTime: String?
//    let companyId: String?
//    let botStatus: Bool?
//    let divisionId: String?
//    let sessionId: String?
//    let sendOutboundFlag: Bool?
//    let closeTime: String?
//    let queueTime: String?
//    let customerAvatar: String?
//    let firstResponseTime: String?
//    let customerEmail: String?
//    let customerName: String?
//    let customerId: String?
//    let windowMessaging: String?
//    let status: Int?
//
//    enum CodingKeys: String, CodingKey {
//        case customerDescription = "customer_description"
//        case lastCustomerMessageTime = "last_customer_message_time"
//        case openTime = "open_time"
//        case channelCode = "channel_code"
//        case customerUsername = "customer_username"
//        case openBy = "open_by"
//        case assignTime = "assign_time"
//        case categories
//        case id
//        case closeBy = "close_by"
//        case handoverBy = "handover_by"
//        case agentUserId = "agent_user_id"
//        case lastAgentChatTime = "last_agent_chat_time"
//        case companyId = "company_id"
//        case botStatus = "bot_status"
//        case divisionId = "division_id"
//        case sessionId = "session_id"
//        case sendOutboundFlag = "send_outbound_flag"
//        case closeTime = "close_time"
//        case queueTime = "queue_time"
//        case customerAvatar = "customer_avatar"
//        case firstResponseTime = "first_response_time"
//        case customerEmail = "customer_email"
//        case customerName = "customer_name"
//        case customerId = "customer_id"
//        case windowMessaging = "window_messaging"
//        case status
//    }
//}
//
//// MARK: - Session
//struct SessionGetConversation: Codable {
//    let roomId: String?
//    let agentUserId: String?
//    let lastAgentChatTime: String?
//    let agent: AgentGetConversation?
//    let botStatus: Bool?
//    let divisionId: String?
//    let openTime: String?
//    let closeTime: String?
//    let closeUser: CloseUserGetConversation?
//    let queueTime: String?
//    let firstResponseTime: String?
//    let sessionPriorityLevelId: String?
//    let openBy: String?
//    let id: String?
//    let categories: String?
//    let assignTime: String?
//    let closeBy: String?
//    let handoverBy: String?
//    let status: Int?
//
//    enum CodingKeys: String, CodingKey {
//        case roomId = "room_id"
//        case agentUserId = "agent_user_id"
//        case lastAgentChatTime = "last_agent_chat_time"
//        case agent
//        case botStatus = "bot_status"
//        case divisionId = "division_id"
//        case openTime = "open_time"
//        case closeTime = "close_time"
//        case closeUser = "close_user"
//        case queueTime = "queue_time"
//        case firstResponseTime = "first_response_time"
//        case sessionPriorityLevelId = "session_priority_level_id"
//        case openBy = "open_by"
//        case id
//        case categories
//        case assignTime = "assign_time"
//        case closeBy = "close_by"
//        case handoverBy = "handover_by"
//        case status
//    }
//}
//
//// MARK: - Agent/User/CloseUser
//struct AgentGetConversation: Codable {
//    let onlineStatus: Int?
//    let customerChannel: String?
//    let description: String?
//    let divisionId: String?
//    let avatar: String?
//    let tags: String?
//    let isBlocked: Bool?
//    let phone: String?
//    let name: String?
//    let id: String?
//    let email: String?
//    let username: String?
//    let status: Int?
//
//    enum CodingKeys: String, CodingKey {
//        case onlineStatus = "online_status"
//        case customerChannel = "customer_channel"
//        case description
//        case divisionId = "division_id"
//        case avatar
//        case tags
//        case isBlocked = "is_blocked"
//        case phone
//        case name
//        case id
//        case email
//        case username
//        case status
//    }
//}
//
//typealias UserGetConversation = AgentGetConversation
//typealias CloseUserGetConversation = AgentGetConversation
//
//// MARK: - AnyCodable (for errors)
//struct AnyCodable: Codable {}
